### PR TITLE
Added Private Routing for in app navigation 

### PR DESCRIPTION
**Pull request by lemohammed**
Created a private route component for routes that require an authenticated state.

### CODE REVIEW

**Review by rjb75 — APPROVED** [review]
Looks good! 🔐

File: src/components/PrivateRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react';
```
Comment (RatikKapoor) [inline src/components/PrivateRoute.tsx:1]: 🤩

File: src/components/PrivateRoute.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    const [loading, setLoading] = useState(true);
+
+    const [authed, setAuthed] = useState<boolean | null>();
+    app.auth().onAuthStateChanged(function (user) {
+        setAuthed(Boolean(user));
+        setLoading(authed == null);
+    });
+
+    console.log(authed);
+    const loadedComponent = authed ? (
```
Comment (RatikKapoor) [inline src/components/PrivateRoute.tsx:27]: 😍

**Review by RatikKapoor — APPROVED** [review]
Amazing job @lemohammed! 🌟

**Pull request merged**